packages/emitter-framework/src/python/components/class-declaration/class-member.tsx

```diff
@@ -0,0 +1,195 @@
+      // when needed. For non-float fields, default to a plain numeric Atom.
+
+      // Unwrap potential numeric wrapper shape and preserve float formatting
+      let raw: any = (defaultValue as any).value;
+      // Example: value is { value: "100", isInteger: true }
+      if (raw && typeof raw === "object" && "value" in raw) raw = raw.value;
+
+      // Float-like property types (including custom subtypes) should render with float hint
+      if (
+        $.scalar.extendsFloat(propertyType) ||
+        $.scalar.extendsFloat32(propertyType) ||
+        $.scalar.extendsFloat64(propertyType) ||
+        $.scalar.extendsDecimal(propertyType) ||
+        $.scalar.extendsDecimal128(propertyType)
+      ) {
+        return <Atom value={defaultValue} assumeFloat />;
+      }
+
+      // Otherwise output as a number atom
+      return <py.Atom jsValue={Number(raw)} />;
+    }
+    case "ArrayValue":
+      return <Atom value={defaultValue} />;
+    default:
+      return undefined;
+  }
+}
+
+/**
+ * Builds the type node for the property. This handles various literal and union variant scenarios:
+ * - Single union variant reference: Color.blue produces Literal[Color.BLUE]
+ * - Union of string literals: "a" | "b" produces Literal["a", "b"]
+ * - Union of integer literals: 1 | 2 | 3 produces Literal[1, 2, 3]
+ * - Union of boolean literals: true | false produces Literal[True, False]
+ * - Union of variant references: Color.red | Color.blue produces Literal[Color.RED, Color.BLUE]
+ * - Mixed literal unions: "a" | 1 | true | Color.RED produces Literal["a", 1, True, Color.RED]
+ * 
+ * @param unpackedType - The unpacked type.
+ * @returns The type node, or undefined if the type doesn't match any supported literal pattern.
+ */
+function buildTypeNodeForProperty(unpackedType: any): Children | undefined {
+  const { $ } = useTsp();
+  
+  // Single union variant reference - Literal[Color.MEMBER]
+  if (unpackedType && unpackedType.kind === "UnionVariant" && unpackedType.union) {
+    const unionType = unpackedType.union;
+    const variantValue = unpackedType.type;
+    const enumMemberName =
+      variantValue && typeof variantValue.value === "string"
+        ? variantValue.value
+        : String(variantValue?.value ?? "");
+    return (
+      <>
+        {typingModule["."]["Literal"]}[{efRefkey(unionType)}.{enumMemberName}]
+      </>
+    );
+  }
+
+  // Union of literals or variant references (including mixed)
+  if (
+    unpackedType &&
+    unpackedType.kind === "Union" &&
+    Array.isArray((unpackedType as any).options)
+  ) {
+    const opts: any[] = (unpackedType as any).options;
+    
+    // Check if all options are valid literal types using typekit
+    const allLiterals = opts.every((opt) => 
+      opt && (
+        $.literal.isString(opt) ||
+        $.literal.isNumeric(opt) ||
+        $.literal.isBoolean(opt) ||
+        opt.kind === "UnionVariant"
```

Comment:
nit: this would be a generally useful "isLiteral" utility function for the python EF